### PR TITLE
feat(topology): scale requests support dry-run

### LINKED ISSUE #15103: Add a dry-run flag for scale request

### PR DESCRIPTION
**Pull request by lenaschoenburg**
This adds support for dry-running scale requests. This is implemented pretty naively by just passing down a boolean `dryRun` flag from the actuator down to the `TopologyManagementRequestsHandler` which decides between applying and just simulating the changes.
To use it, users append a query parameter to the usual request: `POST /actuator/cluster/brokers?dryRun=true`. The response type remains the same and includes current topology, expected topology and planned changes.
If the `dryRun` query parameter is not supplied, we assume no dry run! We have the option to later change the default and only apply changes if users explicitly request `?dryRun=false`.

The way it's implemented now means that we need to support dry runs for all topology operations explicitly. I've also tested out variants where the `dryRun` flag is not part of the request with the goal of automatically supporting dry run of all requests but decided against it because it made the serialization and setup of sender/receiver much more complex.

Closes #15103

### CODE REVIEW

File: topology/src/main/java/io/camunda/zeebe/topology/changes/TopologyChangeCoordinatorImpl.java [hunk 1]
```diff
@@ -65,6 +65,31 @@ public ActorFuture<TopologyChangeResult> applyOperations(final TopologyChangeReq
     return future;
   }
 
+  @Override
+  public ActorFuture<TopologyChangeResult> simulateOperations(final TopologyChangeRequest request) {
+    final ActorFuture<TopologyChangeResult> future = executor.createFuture();
+    executor.run(
```
Comment (deepthidevaki) [inline topology/src/main/java/io/camunda/zeebe/topology/changes/TopologyChangeCoordinatorImpl.java:71]: :question: Can we reuse most of the code in `applyOperations`? Here there is a risk of different behavior in simulation vs actual apply, especially when making changes in future.

File: topology/src/main/java/io/camunda/zeebe/topology/changes/TopologyChangeCoordinatorImpl.java [hunk 2]
```diff
@@ -115,6 +140,26 @@ private void applyOperationsOnTopology(
         });
   }
 
+  private void simulateOperationsOnTopology(
+      final ClusterTopology currentClusterTopology,
+      final List<TopologyChangeOperation> operations,
+      final ActorFuture<TopologyChangeResult> future) {
+    validateTopologyChangeRequest(currentClusterTopology, operations)
```
Comment (deepthidevaki) [inline topology/src/main/java/io/camunda/zeebe/topology/changes/TopologyChangeCoordinatorImpl.java:147]: :x: I think this fails if `operations` is empty. In `apply` we check it and complete the future successfully.

File: dist/src/main/java/io/camunda/zeebe/shared/management/ClusterEndpoint.java [hunk 3]
```diff
@@ -99,7 +106,12 @@ public ResponseEntity<?> add(
 
   @DeleteMapping(path = "/{resource}/{id}")
   public ResponseEntity<?> remove(
-      @PathVariable("resource") final Resource resource, @PathVariable final int id) {
+      @PathVariable("resource") final Resource resource,
+      @PathVariable final int id,
+      @RequestParam(defaultValue = "false") final boolean dryRun) {
```
Comment (deepthidevaki) [inline dist/src/main/java/io/camunda/zeebe/shared/management/ClusterEndpoint.java:111]: :x: Can you add this to the spec?

**Review by deepthidevaki — APPROVED** [review]
:tada:  Please check my comments before merging.

File: topology/src/main/java/io/camunda/zeebe/topology/changes/TopologyChangeCoordinatorImpl.java (hunk 1 above)
Comment (oleschoenburg) [reply topology/src/main/java/io/camunda/zeebe/topology/changes/TopologyChangeCoordinatorImpl.java:55]: Yeah good point, I didn't go far enough the first time. See https://github.com/camunda/zeebe/pull/15170/commits/33bbd719278f93f0488de9becb09692ea2bc12e5, this should be better now,

File: topology/src/main/java/io/camunda/zeebe/topology/changes/TopologyChangeCoordinatorImpl.java (hunk 2 above)
Comment (oleschoenburg) [reply topology/src/main/java/io/camunda/zeebe/topology/changes/TopologyChangeCoordinatorImpl.java]: Fixed by reusing more code between apply and dry-run: https://github.com/camunda/zeebe/pull/15170/commits/33bbd719278f93f0488de9becb09692ea2bc12e5

**Comment by oleschoenburg** [comment]
bors merge

**Pull request merged**